core/envs/griduniverse_env.py

```diff
@@ -117,22 +129,57 @@ def _generate_world(self):
                              for x in np.nditer(np.arange(self.x_max))), dtype='int64, int64')
         return world
 
-    def _generate_walls(self, walls):
+    def _setup_walls(self, walls):
         """
         Given a list of wall indices, fills in self.wall_indices list
         and places "1"s appropriately within self.walls numpy array
 
         self.walls: need index positioning for efficient check in _is_wall() but
         self.wall_indices: we also need list of indices to easily access each wall sequentially (e.g in render())
         """
-        if walls is not None:
-            for wall_state in walls:
+        if not walls:
+            self.initial_walls = []
+        else:
+            self.initial_walls = walls
+            self.wall_grid = np.zeros(self.world.shape)
+            self.wall_indices = []
+            self.non_wall_blocked_states = []
+            for wall_state in self.initial_walls:
+                if not isinstance(wall_state, int):
+                    raise TypeError("Wall state {} is not an integer".format(wall_state))
                 if wall_state < 0 or wall_state > (self.world.size - 1):
                     raise ValueError("Wall state {} is out of grid bounds".format(wall_state))
 
                 self.wall_grid[wall_state] = 1
                 self.wall_indices.append(wall_state)
 
+        self.non_wall_blocked_states = [x for x in range(self.world.size) if x not in self.wall_indices]
+
+    def _setup_levers(self, lever_dict):
+        """
+        Setup levers
```

Comment:
Document how should I input my lever dict. I shouldn't have to look somewhere else to find that info